### PR TITLE
rename testcookie to make it clear it is created by Matomo

### PR DESCRIPTION
**Pull request by Findus23**
inspired by https://forum.matomo.org/t/is-cookie-test-with-value-testvalue-set-by-matomo/40775/2

The cookie's name doesn't start with _pk, so it isn't obvious it is created by Matomo.

I also named the cookie `_pk_testcookie_domain`, so it isn't confused with this testcookie:

https://github.com/matomo-org/matomo/blob/bd7e099e46de3e10afc069624b713de4bd8df162/js/piwik.js#L3043-L3044

### Review

* [x] Functional review done
* [x] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [x] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [x] Code review done
* [x] Tests were added if useful/possible
* [x] Reviewed for breaking changes
* [x] Developer changelog updated if needed
* [x] Documentation added if needed
* [x] Existing documentation updated if needed

### CODE REVIEW

**Comment by Findus23** [comment]
build js

**Review by flamisz — APPROVED** [review]
LGTM

File: js/piwik.js [hunk 1]
```diff
@@ -3381,8 +3381,9 @@ if (typeof window.Matomo !== 'object') {
 
             function isPossibleToSetCookieOnDomain(domainToTest)
             {
+                var testCookieName = configCookieNamePrefix + 'testcookie_domain';
                 var valueToSet = 'testvalue';
-                setCookie('test', valueToSet, 10000, null, domainToTest, configCookieIsSecure, configCookieSameSite);
+                setCookie(testCookieName, valueToSet, 10000, null, domainToTest, configCookieIsSecure, configCookieSameSite);
 
                 if (getCookie('test') === valueToSet) {
                     deleteCookie('test', null, domainToTest);
```
Comment (sgiehl) [inline js/piwik.js]: shouldn't that use the new `testCookieName` variable as well? 🤔

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: js/piwik.js (hunk 1 above)
Comment (Findus23) [reply js/piwik.js]: You are of course right :+1:

**Comment by Findus23** [comment]
build js

**Review by sgiehl — APPROVED** [review]

**Pull request merged**